File: src/main/java/com/liztube/service/AuthService.java
package com.liztube.service;

import com.liztube.business.AuthBusiness;
import com.liztube.exception.ServiceException;
import com.liztube.exception.SigninException;
import com.liztube.exception.UserNotFoundException;
import com.liztube.exception.exceptionType.PublicException;
import com.liztube.utils.facade.TestExistFacade;
import com.liztube.utils.facade.UserConnectedProfile;
import com.liztube.utils.facade.UserForRegistration;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

/**
 * Service which provide all the method concerning the authentication of an user (login, signin, get user profile...)
 */
@RestController
@RequestMapping("/api/auth")
public class AuthService {
    @Autowired
    AuthBusiness authBusiness;

    /**
     * Get current user profile
     * @return
     * @throws UserNotFoundException
     */
    @RequestMapping(value = "/currentProfil",method = RequestMethod.GET)
    public UserConnectedProfile getUserProfil() throws ServiceException {
        try{
            return authBusiness.getUserConnectedProfile(true);
        }catch (Exception e){
            e.printStackTrace();
            throw new ServiceException("Get connected user profile");
        }
    }

    /**
     * Register to the liztube service as a user
     * @param userForRegistration
     * @return
     * @throws SigninException
     */
    @RequestMapping(value = "/signin", method = RequestMethod.POST)
    public long signIn(@RequestBody UserForRegistration userForRegistration) throws SigninException, ServiceException {
        try{
            return authBusiness.signIn(userForRegistration).getId();
        }catch (PublicException e){
            throw e;
        }catch (Exception e){
            e.printStackTrace();
            throw new ServiceException("Sign in");
        }
    }

    /**
     * Determine if a pseudo is already used
     * @param testExistFacade
     * @return
     */
    @RequestMapping(value = "/pseudo", method = RequestMethod.POST)
    public Boolean existPseudo(@RequestBody TestExistFacade testExistFacade) throws ServiceException {
        try{
            return authBusiness.existPseudo(testExistFacade);
        }catch (Exception e){
            e.printStackTrace();
            throw new ServiceException("Pseudo exists");
        }
    }

    /**
     * Determine if an email is already used
     * @param testExistFacade
     * @return
     */
    @RequestMapping(value = "/email", method = RequestMethod.POST)
    public Boolean existEmail(@RequestBody TestExistFacade testExistFacade) throws ServiceException {
        try{
            return authBusiness.existEmail(testExistFacade);
        }catch (Exception e){
            e.printStackTrace();
            throw new ServiceException("Email exists");
        }
    }


}

File: FrontEndDev/resources/common/dataService/authService.js
/**
Authentication data service : to get/set data from the API

**/

angular.module('liztube.dataService.authService', [
    'restangular'
]).factory('authService', function (Restangular) {
    var RestangularDefault = Restangular.withConfig(function(RestangularConfigurer) {
        RestangularConfigurer.setBaseUrl('/');
    });

    function baseAuth(){
        return Restangular.one("auth");
    }

    return {
        login: login,
        register: register,
        currentUser: currentUser,
        logout: logout,
        emailExist : emailExist,
        pseudoExist : pseudoExist
    };

    /**
    Get current connected user
    **/
    function currentUser() {
        return baseAuth().one('currentProfil').get();
    }

    function login(username, password) {
        return RestangularDefault.one('login')
        .customPOST("username="+username+"&password="+password,
            undefined,
            undefined,
            {'Content-Type': 'application/x-www-form-urlencoded'});
    }
    function register(user) {
        return baseAuth().post('signin', user);
    }
    function emailExist(email) {
        var emailObj = {
            'value':email
        };
        return baseAuth().post('email', emailObj);
    }
    function pseudoExist(pseudo) {
        var pseudoObj = {
            'value':pseudo
        };
        return baseAuth().post('pseudo', pseudoObj);
    }
    function logout(){
        return RestangularDefault.one('logout').get();
    }
});